app/controllers/api_controller/git_repositories.rb

```diff
@@ -0,0 +1,31 @@
+class ApiController
+  module GitRepositories
+    CREDENTIALS_ATTR = 'credentials'.freeze
+    VERIFY_SSL_ATTR = 'verify_ssl'.freeze
+    SUPPORTED_CRED_ATTRS = %w(userid password).freeze
+
+    def create_resource_git_repositories(_type, _id, data = {})
+      klass = collection_class(@req[:collection])
+      git_repository_data = data.except(CREDENTIALS_ATTR)
+      git_repository_data.merge!(data[CREDENTIALS_ATTR].slice(VERIFY_SSL_ATTR)) if data[CREDENTIALS_ATTR]
+      invalid_keys = git_repository_data.keys - klass.columns_hash.keys
+      raise BadRequestError, "Invalid attributes #{invalid_keys.join(', ')} specified" if invalid_keys.present?
```

Comment:
We already have a check for this in place. (https://github.com/ManageIQ/manageiq/blob/ea18c066109219d0c32faebcc9f54722bbad9e14/app/controllers/api_controller/manager.rb#L139)

It seems that when id or href are supplied, we treat the request as an update, and then complain if the action is not the update.

Let's see if I can write a test for this. So we don't loose this with future refactorings.